coderd/workspaceapps/apptest/setup.go

```diff
@@ -58,10 +58,14 @@ type DeploymentOptions struct {
 type Deployment struct {
 	Options *DeploymentOptions
 
-	// Client should be logged in as the admin user.
-	Client         *codersdk.Client
+	// APIClient should be logged in as the admin user.
+	APIClient      *codersdk.Client
 	FirstUser      codersdk.CreateFirstUserResponse
 	PathAppBaseURL *url.URL
+
+	// AppHostServesAPI is true if the app host is also the API server. This
+	// disables any tests that test API passthrough.
+	AppHostServesAPI bool
```

Comment:
This is so we can conditionally disable tests that don't work unless the server that is serving apps in the test is the same as the API server. It's set to true by the deployment factory if the deployment it made was a API + moon deployment or just an API deployment.

Tests that are different include auth tests (because path apps don't ever smuggle keys on the primary).

It's not a "please make a server that serves API" flag, it's a "I just made a server that serves an API" flag. Output, not input

Reply:
I see.

Since it's primary usage is for tests, can we name it as such? It'd help me understand future-forward.

Reply:
The whole package is for tests... but I can improve the comment